config/config.go

```diff
@@ -172,17 +175,20 @@ func (c *Config) Ensure(fromCloud bool, logger logging.Logger) error {
 	}
 
 	for idx := 0; idx < len(c.Services); idx++ {
-		dependsOn, err := c.Services[idx].Validate(fmt.Sprintf("%s.%d", "services", idx), resource.APITypeServiceName)
+		service := c.Services[idx]
+		dependsOn, err := service.Validate(fmt.Sprintf("%s.%d", "services", idx), resource.APITypeServiceName)
 		if err != nil {
 			if c.DisablePartialStart {
 				return err
 			}
-			logger.Errorw("service config error; starting robot without service", "name", c.Services[idx].Name, "error", err)
+			logger.Errorw("service config error; starting robot without service", "name", service.Name, "error", err)
+			resLogger := logger.Sublogger(service.ResourceName().String())
```

Comment:
Same noise pattern. Factored out the `c.Services[idx]` bit. This line and following are the real change.